src/normalizer/chinese.rs

```diff
@@ -1,26 +1,47 @@
 use std::borrow::Cow;
 
-use character_converter::traditional_to_simplified;
+use pinyin::ToPinyin;
 
 use super::{Normalizer, NormalizerOption};
 use crate::detection::{Language, Script};
 use crate::Token;
 
-/// Normalize Chinese characters by converting them into Simplified Chinese characters.
+/// Normalize Chinese characters by converting them into Pinyin characters.
 ///
-/// This Normalizer uses [`character_converter`] internally to normalize the provided token.
+/// This Normalizer uses [`pinyin`] internally to normalize the provided token.
 pub struct ChineseNormalizer;
 
 impl Normalizer for ChineseNormalizer {
     fn normalize<'o>(
         &self,
         mut token: Token<'o>,
-        _options: NormalizerOption,
+        options: NormalizerOption,
     ) -> Box<dyn Iterator<Item = Token<'o>> + 'o> {
-        if let Cow::Owned(s) = traditional_to_simplified(token.lemma()) {
-            token.lemma = Cow::Owned(s);
+        let mut lemma = String::new();
+
+        // Need to create char_map before converting into Pinyin
+        if options.create_char_map {
+            let mut char_map = Vec::new();
+
+            for c in token.lemma().chars() {
+                let char_len = c.len_utf8() as u8;
+
+                char_map.push((char_len, char_len));
```

Comment:
the role of the char map is to bind the original char length with its normalized version length,
the char map has to be created at the same time as the normalization.
I suggest merging this for loop with the one below.
Creating a Vec has no cost if you never push in it.
In another hand, you can have an example of how the fill the char map in [the Latin normalizer](https://github.com/meilisearch/charabia/blob/99b713ce0b70e733611d7c1446be48bca7bb19de/src/normalizer/latin.rs#L20-L35).

Reply:
Thank you for your review - the char map role was a bit unclear for me, but after your explanation I finally got it (:

All the changes should be addressed now.